recipes/protobuf/all/conanfile.py

```diff
@@ -45,32 +46,58 @@ def configure(self):
             if compiler_version < "14":
                 raise ConanInvalidConfiguration("On Windows Protobuf can only be built with "
                                                 "Visual Studio 2015 or higher.")
+        if tools.is_apple_os(self.settings.os) and self.options.shared:
+            raise ConanInvalidConfiguration("Protobuf could not be build as shared library for Mac.")
+        if self.options.lite:
+            del self.options.with_zlib
 
     def requirements(self):
-        if self.options.with_zlib:
+        if self.options.get_safe("with_zlib"):
             self.requires("zlib/1.2.11")
 
     @property
     def _cmake_install_base_path(self):
         return os.path.join("lib", "cmake", "protobuf")
 
     def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["CMAKE_INSTALL_CMAKEDIR"] = self._cmake_install_base_path.replace("\\", "/")
-        self._cmake.definitions["protobuf_BUILD_TESTS"] = False
-        self._cmake.definitions["protobuf_WITH_ZLIB"] = self.options.with_zlib
-        self._cmake.definitions["protobuf_BUILD_PROTOC_BINARIES"] = not self.options.lite
-        self._cmake.definitions["protobuf_BUILD_PROTOBUF_LITE"] = self.options.lite
-        if self.settings.compiler == "Visual Studio":
-            self._cmake.definitions["protobuf_MSVC_STATIC_RUNTIME"] = "MT" in self.settings.compiler.runtime
-        self._cmake.configure(build_folder=self._build_subfolder)
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["CMAKE_INSTALL_CMAKEDIR"] = self._cmake_install_base_path.replace("\\", "/")
+            self._cmake.definitions["protobuf_BUILD_TESTS"] = False
+            self._cmake.definitions["protobuf_WITH_ZLIB"] = self.options.get_safe("with_zlib", False)
+            if self.settings.compiler == "Visual Studio":
+                self._cmake.definitions["protobuf_MSVC_STATIC_RUNTIME"] = "MT" in str(self.settings.compiler.runtime)
+            self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
     def build(self):
-        for patch in self.conan_data["patches"][self.version]:
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "cmake", "protobuf-config.cmake.in"),
+            "@_protobuf_FIND_ZLIB@",
+            "# CONAN PATCH _protobuf_FIND_ZLIB@"
+        )
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "cmake", "protobuf-config.cmake.in"),
+            """COMMAND  protobuf::protoc
+      ARGS --${protobuf_generate_LANGUAGE}_out ${_dll_export_decl}${protobuf_generate_PROTOC_OUT_DIR} ${_protobuf_include_path} ${_abs_file}
```

Comment:
There's a problem with this line. Different versions have different content. 
* v3.9.1 has: `ARGS --${protobuf_generate_LANGUAGE}_out ${_dll_export_decl}${protobuf_generate_PROTOC_OUT_DIR} ${_protobuf_include_path} ${_abs_file}`
* v3.11.4 has:  `ARGS --${protobuf_generate_LANGUAGE}_out ${_dll_export_decl}${protobuf_generate_PROTOC_OUT_DIR} ${_protobuf_include_path} ${_abs_file}`
* v3.12.4 has: `ARGS --${protobuf_generate_LANGUAGE}_out ${_dll_export_decl}${protobuf_generate_PROTOC_OUT_DIR} ${_plugin} ${_protobuf_include_path} ${_abs_file}`

Notice the extra `${_plugin}` on v3.12.4.

Reply:
@Croydon 
Has this problem been resolved?

Reply:
Yes